### PR TITLE
feat: API access

### PR DESCRIPTION
**Pull request by dnsos**
This PR connects the frontend to the API. We access the API at two points:

### In a store: `src/state/store.ts`
Here we load all devices, group them by project and connect some additional (currently hard-coded) information.

### In the `src/components/Project.tsx`
This component gets rendered whenever we access the detail page of a project. Here we request additional data from the API (currently only the records of the associated devices).

In addition, this PR updates some data passing, formatting, etc.

Closes DATAHUB-39

### CODE REVIEW

File: src/lib/requests.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+export interface DeviceResponse {
+  data: {
+    devices: {
```
Comment (ff6347) [inline src/lib/requests.ts]: Awesome. Your getting into TS. :) 

I suggest splitting interfaces so You have the DeviceResponse and devices is from the Type Device []

e.g.
```typescript
interface DeviceResponse{
	data: {
	devices: Device[];
	}
}

interface Device {
	id: string;
	// …
}
```

Comment (ff6347) [reply src/lib/requests.ts]: Just a question. Does the API return devices as Object or as Array of Device?

Comment (dnsos) [reply src/lib/requests.ts]: As an array of devices. I might still be messing up some type declarations, let me know then ;) But also, there will be a few more commits in this PR. I would ask you to take a look later today

Comment (ff6347) [reply src/lib/requests.ts]: Sure. I'll give it a test ride later on.

**Pull request merged**